File: server/gameserver/src/me/eddiep/ghost/gameserver/packets/OkPacket.java
package me.eddiep.ghost.gameserver.packets;

import me.eddiep.ghost.gameserver.TcpUdpClient;
import me.eddiep.ghost.gameserver.TcpUdpServer;
import me.eddiep.ghost.network.packet.Packet;

import java.io.IOException;

public class OkPacket extends Packet<TcpUdpServer, TcpUdpClient> {
    public OkPacket(TcpUdpClient client) {
        super(client);
    }

    @Override
    protected void onWritePacket(TcpUdpClient client, Object... args) throws IOException {
        if (args.length == 0)
            return;

        boolean isOk = (boolean)args[0];

        write((byte)0x01)
                .write(isOk)
                .endTCP();
    }
}

File: client/core/src/me/eddiep/ghost/client/network/packets/OKPacket.kt
package me.eddiep.ghost.client.network.packets

import me.eddiep.ghost.client.network.Packet
import me.eddiep.ghost.client.network.PlayerClient

class OKPacket() : Packet<PlayerClient>() {

    override fun handle() {
        val isOk : Boolean = consume(1).asBoolean()
    }
}
